images/osm2pgsql/run.py

```diff
@@ -51,123 +112,143 @@ def extract_integer_value(content, find, throw_not_found=True, none_found=-1):
     try:
         found = int(found)
     except ValueError:
-        raise ValueError(f'"{find}=" must be follow by a positive integer at state.txt')
+        raise ValueError(
+            f'"{find}=" must be follow by a positive integer at state.txt')
 
     return found
 
+
 def has_data():
     flag = True
     while (flag):
         try:
-            conn = psycopg2.connect(f"host={PGHOST} port={PGPORT} dbname={PGDATABASE} user={PGUSER} password={PGPASSWORD}")
+            conn = psycopg2.connect(
+                f"host={PGHOST} port={PGPORT} dbname={PGDATABASE} user={PGUSER} password={PGPASSWORD}")
             cur = conn.cursor()
-            cur.execute("SELECT count(*) FROM information_schema.tables WHERE table_schema = 'public'")
+            cur.execute(
+                "SELECT count(*) FROM information_schema.tables WHERE table_schema = 'public'")
             data = cur.fetchone()[0]
             flag = False
             return data
+        # should it do this or just crash?
         except:
-            time.sleep(os.environ['OSM2PGSQL_UPDATE_INTERVAL'])
+            time.sleep(UPDATE_INTERVAL)
 
-def update_replication_tiles(sequence_number, last_append):
-    for i in range(last_append+1, sequence_number):
+
+def update_db_with_replication(api_db_sequence_number, tiler_db_sequence_number):
+    for i in range(tiler_db_sequence_number+1, api_db_sequence_number+1):
         dir1 = parse_integer_to_directory_number(int(i / divide_for_days))
         dir2 = parse_integer_to_directory_number(int(i / divide_for_month))
         state = parse_integer_to_directory_number(int(i % divide_for_years))
 
-        dir_path = os.path.join(expired_directory, dir1, dir2)
+        # creating the folder to store the expired tiles list if it does not exist
+        if not is_file_or_directory('/'.join([EXPIRED_DIRECTORY, dir1])):
+            os.mkdir('/'.join([EXPIRED_DIRECTORY, dir1]))
+
+        EXPIRED_PATH = '/'.join([EXPIRED_DIRECTORY, dir1, dir2])
+        if not is_file_or_directory(EXPIRED_PATH):
+            os.mkdir(EXPIRED_PATH)
+
+        # downloading the osc file
+        download('/'.join([dir1, dir2, state + '.osc.gz']), DOWNLOAD_DIR)
+
+        osc = os.path.join(DOWNLOAD_DIR, state + ".osc.gz")
```

Comment:
let's extract the file extension of `.osc.gz` to a global value

Reply:
fixed